x/tokenfactory/keeper/before_send.go

```diff
@@ -99,6 +99,11 @@ func (k Keeper) callBeforeSendListener(ctx context.Context, from, to sdk.AccAddr
 				return err
 			}
 
+			// Do not invoke hook if denom is not whitelisted
+			if err := k.AssertIsHookWhitelisted(c, coin.Denom, cwAddr); err != nil {
+				return nil
```

Comment:
why `return nil`? new hooks can't be set if not whietlisted, would make sense to return an error here imo